feat: sorted completed quests and created daily and main quest routes #1

fishonamos opened the pull request:
- [x] Follow the[ contribution guidelines](https://github.com/keep-starknet-strange/art-peace/blob/main/CONTRIBUTING.md)
- [x] On the Quests.js file add sorting for completed quests 
- [x] Create Backend Route to Query Quests from Postgres
- [x] Expand route for dailyQuests and mainQuests.
- [x]  Request mainQuests and dailyQuests route on the frontend  Quest component.


This will solve #13

b-j-roberts commented:
Hey, thanks for the draft. Looking good so far!

b-j-roberts commented on backend/core/quest.go:
```diff
@@ -0,0 +1,11 @@
+package core
+
+// Quest represents a quest from the database
+type Quest struct {
```
Hey, there have been some changes in the quests setup since creating the issue.

There are 2 quest types, `DailyQuests` and `MainQuests`. If you could expand this implementation to accommodate those, that'd be awesome. Sorry.

- [Daily Quests](https://github.com/keep-starknet-strange/art-peace/blob/3125497b8e9879fe63ddb7b96c2c1818489bc0c6/postgres/init.sql#L44)
- [Main Quests](https://github.com/keep-starknet-strange/art-peace/blob/3125497b8e9879fe63ddb7b96c2c1818489bc0c6/postgres/init.sql#L65)

b-j-roberts commented on backend/routes/Quests.go:
```diff
@@ -0,0 +1,36 @@
+)
+
+// GetQuests handler and also retrieves all quests from the database
+func GetQuests(w http.ResponseWriter, r *http.Request) {
```
Changing this to 2 separate `GetDailyQuests` and `GetMainQuests` functions and routes would be awesome, if you can.

b-j-roberts commented on backend/routes/routes.go:
```diff
@@ -1,11 +1,17 @@
-	InitUserRoutes()
-	InitContractRoutes()
+import (
+  "github.com/gorilla/mux"
```
I'm curious about switching to `gorilla/mux.` Does it help a lot with your issue, or is it just a change you think will be useful in the future?

b-j-roberts commented on backend/routes/routes.go:
```diff
@@ -1,11 +1,17 @@
+)
+
+func InitRoutes(router *mux.Router) { // THis will accepts the router instance as an argument
+  InitIndexerRoutes(router)
```
I think you'll need to change all these route functions if we decide to swap to `gorilla/mux`

b-j-roberts reviewed (changes requested):
Let me know if you'll be able to make these changes. Thank you for the work so far!

fishonamos commented:
Hi @b-j-roberts Hi there,

I made some changes and modifications to accommodate the new DailyQuests and MainQuests. 

In art-peace/backend/core/quest.go, I have created two separate structs DailyQuest and MainQuest to represent the corresponding Postgres tables. 

I have also updated the Quest interface to accommodate these new change.
In art-peace/backend/routes/Quests.go, I have removed the gorilla/mux package and created two separate handler functions GetDailyQuests and GetMainQuests for the two quests. I have also updated the InitQuestsRoutes function to register these new handler functions.

b-j-roberts commented:
Nice work by the way, and thanks for making that change!
